src/core/server/http/types.ts

```diff
@@ -36,3 +42,200 @@ export type RequestHandlerContextContainer = IContextContainer<RequestHandler<an
+ *   }),
+ * };
+ *
+ * router.get({
+ *   path: 'path/{id}',
+ *   validate
+ * },
+ * async (context, request, response) => {
+ *   const data = await findObject(request.params.id);
+ *   if (!data) return response.notFound();
+ *   return response.ok({
+ *     body: data,
+ *     headers: {
+ *       'content-type': 'application/json'
+ *     }
+ *   });
+ * });
+ * ```
+ * @public
+ */
+export interface HttpServiceSetup {
+  /**
+   * Creates cookie based session storage factory {@link SessionStorageFactory}
+   * @param cookieOptions {@link SessionStorageCookieOptions} - options to configure created cookie session storage.
+   */
+  createCookieSessionStorageFactory: <T>(
+    cookieOptions: SessionStorageCookieOptions<T>
+  ) => Promise<SessionStorageFactory<T>>;
+
+  /**
+   * To define custom logic to perform for incoming requests.
+   *
+   * @remarks
+   * Runs the handler before Auth interceptor performs a check that user has access to requested resources, so it's the
+   * only place when you can forward a request to another URL right on the server.
+   * Can register any number of registerOnPostAuth, which are called in sequence
+   * (from the first registered to the last). See {@link OnPreAuthHandler}.
+   *
+   * @param handler {@link OnPreAuthHandler} - function to call.
+   */
+  registerOnPreAuth: (handler: OnPreAuthHandler) => void;
+
+  /**
+   * To define custom authentication and/or authorization mechanism for incoming requests.
+   *
+   * @remarks
+   * A handler should return a state to associate with the incoming request.
+   * The state can be retrieved later via http.auth.get(..)
+   * Only one AuthenticationHandler can be registered. See {@link AuthenticationHandler}.
+   *
+   * @param handler {@link AuthenticationHandler} - function to perform authentication.
+   */
+  registerAuth: (handler: AuthenticationHandler) => void;
+
+  /**
+   * To define custom logic to perform for incoming requests.
+   *
+   * @remarks
+   * Runs the handler after Auth interceptor
+   * did make sure a user has access to the requested resource.
+   * The auth state is available at stage via http.auth.get(..)
+   * Can register any number of registerOnPreAuth, which are called in sequence
+   * (from the first registered to the last). See {@link OnPostAuthHandler}.
+   *
+   * @param handler {@link OnPostAuthHandler} - function to call.
+   */
+  registerOnPostAuth: (handler: OnPostAuthHandler) => void;
+
+  /**
+   * Access or manipulate the Kibana base path
+   * See {@link IBasePath}.
+   */
+  basePath: IBasePath;
+
+  /**
+   * Flag showing whether a server was configured to use TLS connection.
+   */
+  isTlsEnabled: boolean;
+
+  /**
+   * Provides ability to declare a handler function for a particular path and HTTP request method.
+   *
+   * @remarks
+   * Each route can have only one handler functions, which is executed when the route is matched.
+   * All routes are prefixed with plugin name as a first segment of URL path.
```

Comment:
```suggestion
   * All routes are prefixed with the plugin id as the first segment of the URL path.
```

Not sure if it's a regression but in my integration test this doesn't seem to be the behaviour: https://github.com/elastic/kibana/pull/44143/files#diff-b7014e315f7fb2ece6a93fdecea33074R26

Reply:
Yeah this was removed in https://github.com/elastic/kibana/pull/44855. I'll remove this line for now.